packages/models/src/models/Rooms.ts

```diff
@@ -2206,4 +2206,12 @@ export class RoomsRaw extends BaseRaw<IRoom> implements IRoomsModel {
 			{ returnDocument: 'after' },
 		);
 	}
+
+	markRolePrioritesCreatedForRoom(rid: IRoom['_id']) {
+		return this.updateOne({ _id: rid }, { $set: { rolePrioritiesCreated: true } });
+	}
+
+	async hasCreatedRolePrioritiesForRoom(rid: IRoom['_id']) {
+		return (await this.findOne({ _id: rid, rolePrioritiesCreated: true })) !== null;
```

Comment:
Just return the `findOne` and let the caller do the condition check so you don't await and return.

Reply:
(Or just run the `count` with the query and check it on the caller)